### ISSUE TITLE
Navigation: setting a dashboard as your home page is confusing

### ISSUE THREAD

**Issue by mem** [grafana/grafana#58999]
With the new navigation, setting a dashboard as your home page is confusing.

I have set a particular dashboard as my home page. Opening Grafana or hitting the Grafana logo takes me to the correct dashboard:

![nav-1](https://user-images.githubusercontent.com/189701/202857725-c3eade5b-3d24-4f81-a492-853eef18a3c2.png)

notice it says "Home".

From this dashboard I would like to navigate to another dashboard in the same folder:

If I navigate to a different dashboard in the same folder, I see this:

![nav-2](https://user-images.githubusercontent.com/189701/202858038-f6468731-a262-48b0-aeb3-6d70d0accd7d.png)

Notice I see the link to the folder there.

If I navigate to the dashboard in that folder that is set as the home for Grafana, I briefly see the above, and then I get redirected to the same dashboard under a different URL that looks like the first one.

Effectively, I cannot use the top bar with the folder structure to navigate from Home to other dashboards in that folder. In other words, I don't have a link to the folder that contains the home dashboard. The workaround is to include a link to the folder in the dashboard's links. And I have to do that for every single dashboard in the folder if I want consistency (because now I have the link in one place for all dashboards but one, and in a different place for that one dashboard).

For a while the redirect mentioned in the second point wasn't happening. So I could go to that dashboard using navigation. In that case the annoying part was the behavior of the home button, because it would be the same dashboard with a different navigation bar.

**Comment by ashharrison90** [comment]
@mem wanted to give an update so you're aware we're not ignoring this, just collecting information for now.

there's currently a couple of ways to set a home dashboard:

1. setting via config
    - using `default_home_dashboard_path = <pathToDashboardJson>`
    - this modifies the backend such that the root url (usually `/`) serves up that home page/dashboard.
    - no url parsing required! if you're on your root url, you're home
1. setting via the preferences page
    - this adds a redirect on that home route (`/`) to the dashboard route (`/d/<something>`)
    - because of this, we have to do some url parsing in the breadcrumb component and basically check if the url matches the home url. this is why when you navigate to that dashboard via browse/search page, it will also just show `Home` in the breadcrumbs - because the url matches 😬 
    
think the ideal behaviour would be that however you set your home dashboard, it is served up by the root url and there are no redirections at all. that way, navigating to a dashboard via the browse/search experience will always show the full breadcrumb tree, and clicking `Home` will always show just the `Home` breadcrumb.

**but** i'm not sure if that's possible/if there are any technical limitations preventing us from doing that. it would also be a fairly large piece of refactoring. 

in the mean time, i think the behaviour we've got is the least bad option. it's definitely annoying not being able to navigate directly to the folder, but we found in initial user tests that it was much more confusing when the `Home` breadcrumb wasn't behaving as expected.
